frontend/rockcraft.yaml

```diff
@@ -23,45 +22,50 @@ parts:
     override-build: |
       mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
       (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  # This part builds the client side of the frontend container image
   frontend:
     plugin: npm
     source: https://github.com/kubeflow/pipelines.git
-    source-tag: 2.0.0-alpha.7
+    source-tag: 2.0.0
     build-snaps:
     - node/14/stable
     override-build: |
+        # Change working directory to pipelines/frontend
         cd frontend
-        npm ci
-        npm run postinstall
+        npm ci && npm run postinstall
         npm run build
+        # Create a client directory in root and copy the build directory into it
         mkdir -p ${CRAFT_PART_INSTALL}/client
-        cp -r build ${CRAFT_PART_INSTALL}/client
+        cp -a build ${CRAFT_PART_INSTALL}/client
+
+  # This part builds the server side of the frontend container image
   backend:
     plugin: npm
     source: https://github.com/kubeflow/pipelines.git
-    source-tag: 2.0.0-alpha.7
+    source-tag: 2.0.0
     build-snaps:
     - node/14/stable
     override-build: |
-        cd frontend/server
-        mkdir -p dist
+        # Change working directory to pipelines/frontend
+        cd ./frontend
+        mkdir -p ./server/dist
         export BUILD_VERSION=$(git describe --abbrev=0 --tags)
         export BUILD_COMMIT=$(git rev-parse HEAD)
         export BUILD_DATE=$(date "+%F-%H-%M-%S")
-        echo $BUILD_COMMIT > ./dist/COMMIT_HASH
-        echo $BUILD_DATE > ./dist/BUILD_DATE
-        echo $BUILD_VERSION > ./dist/TAG_NAME
-        # previous ./scripts/yarn-licenses.sh
-        npm install -g yarn
-        npx yarn install
-        npx yarn licenses generate-disclaimer > ../server-dependency-licenses.txt
+        echo $BUILD_COMMIT > ./server/dist/COMMIT_HASH
+        echo $BUILD_DATE > ./server/dist/BUILD_DATE
+        echo $BUILD_VERSION > ./server/dist/TAG_NAME
+
+        # Generate the dependency licenses files (one for the UI and one for the webserver),
+        # concatenate them to one file under ./src/server
+        ./scripts/yarn-licenses.sh
+
+        # Create a server directory in root and copy the server directory into it
+        cp -r server ${CRAFT_PART_INSTALL}/
+        cd ${CRAFT_PART_INSTALL}/server/
         npm ci
```

Comment:
Something is not right here. Why are we bulding after copying contents?

Reply:
The previous build is from `pipelines/frontend`, this one is from `./src/frontend/server ` which is copied in `/server`.

That's what upstream is doing in their [Dockerfile](https://github.com/kubeflow/pipelines/blob/2.0.0/frontend/Dockerfile#L36).

Reply:
Ok. I see. We build in installation directory.